crates/circuit/src/operations.rs

```diff
@@ -2627,6 +2789,33 @@ impl Operation for PyGate {
     fn directive(&self) -> bool {
         false
     }
+
+    fn get_mat_static_1q(&self, _params: &[Param]) -> Option<[[Complex64; 2]; 2]> {
+        Python::with_gil(|py| -> Option<[[Complex64; 2]; 2]> {
+            match self.num_qubits() {
+                1 => match self.gate.getattr(py, intern!(py, "to_matrix")) {
+                    Ok(to_matrix) => {
+                        let res: Option<PyObject> = to_matrix.call0(py).ok()?.extract(py).ok();
+                        match res {
+                            Some(x) => {
+                                let array: PyReadonlyArray2<Complex64> = x.extract(py).ok()?;
+                                let arr = array.as_array();
+                                Some([[arr[[0, 0]], arr[[0, 1]]], [arr[[1, 0]], arr[[1, 1]]]])
+                            }
+                            None => None,
+                        }
+                    }
+                    Err(_) => None,
+                },
+                _ => None,
+            }
+        })
+    }
+
+    fn get_mat_nalgebra_1q(&self, params: &[Param]) -> Option<Matrix2<Complex64>> {
```

Comment:
Could this just be implemented in `Operation` itself to avoid repeating it in every enum? Something like
```
fn get_mat_nalgebra_1q(...) -> Option< ... > {
    self.get_mat_static_1q(...).map(... your map to Matrix2 ...)
```
Since `map` will leave `None` as `None` that should also work for PyOperation etc, I think